odoo/addons/test_http/tests/test_static.py

```diff
@@ -254,6 +256,108 @@ def test_static15_public_user_image(self):
+        with file_open('web/static/img/nologo.png', 'rb') as file:
+            cls.logo_no_logo_data = file.read()
+        cls.logo_no_logo_data_b64 = base64.encodebytes(cls.logo_no_logo_data)
+        cls.headers_default_logo = {
+            'Content-Length': f'{len(cls.default_logo_data)}',
+            'Content-Type': 'image/png',
+            'Content-Disposition': 'inline; filename=logo.png'
+        }
+        cls.headers_logo_gizeh = {
+            'Content-Length': f'{len(cls.logo_gizeh_data)}',
+            'Content-Type': 'image/png',
+            'Content-Disposition': 'inline; filename=logo.png'
+        }
+        cls.headers_logo_no_logo = {
+            'Content-Length': f'{len(cls.logo_no_logo_data)}',
+            'Content-Type': 'image/png',
+            'Content-Disposition': 'inline; filename=nologo.png'
+        }
+        super_user = cls.env['res.users'].browse([odoo.SUPERUSER_ID])
+        companies = ResCompany.browse([super_user.company_id.id]) | ResCompany.create(
+            {
+                'name': 'Company 2',
+                'email': 'company.2@test.example.com',
+                'country_id': cls.env.ref('base.fr').id,
+            }
+        )
+        cls.company_of_superuser, cls.company2 = companies
+        cls.password = 'Pl1bhD@2!kXZ'
+        cls.user_of_company_of_superuser, cls.user_company2 = [
+            new_test_user(cls.env, f'user_{company.id}', company_id=company.id, password=cls.password)
+            for company in companies]
+
+    def assertDownloadLogo(self, headers, img_data, user=None, company=None):
+        """Assert that the logo endpoint returns the right image and headers.
+
+        :param dict headers: expected headers
+        :param bytes img_data: expected image data
+        :param user: optional user, if set the check will be done while being authenticated with that user
+        :param company: optional company, the company will be appended in the URL parameters
+        :param all_endpoints: whether to test all endpoint or only the main one
+        """
+        url_suffix = f'?company={company.id}' if company else ''
+        if user:
+            self.authenticate(user.login, self.password)
+        else:
+            self.authenticate(None, None)
+        self.assertDownload(f'/logo.png{url_suffix}', 200, headers, img_data)
+
+    def assertDownloadLogoDefault(self, user=None, company=None):
+        self.assertDownloadLogo(self.headers_default_logo, self.default_logo_data, user, company)
+
+    def assertDownloadLogoGizeh(self, user=None, company=None):
+        self.assertDownloadLogo(self.headers_logo_gizeh, self.logo_gizeh_data, user, company)
+
+    def assertDownloadLogoNoLogo(self, user=None, company=None):
+        self.assertDownloadLogo(self.headers_logo_no_logo, self.logo_no_logo_data, user, company)
+
+    def test_company_logo(self):
+        # company_of_superuser: default logo, company2: default logo
```

Comment:
Instead of comments, you can use sub-tests

```suggestion
        with self.subTest(company1="default logo", company2="default logo"):
            ...
```

This serves a double purpose:

1. It acts like a comment in the source code with a precise delimitation (thanks to the with block) of where it begins and where it ends.
2. In case one of the test fails, unittest will report the keyword arguments you used (here: company1="default logo", company2="default logo") so it is easier to spot what's wrong.

But frankly, in this specific case I would not use sub test but write multiple dedicated tests. Each time re-configuring explicitly the companies from scratch (don't rely on the default values set in the `setUpClass`). This would make it very explicit what the configuration of each test is, voiding the need for a comment.

Reply:
Thanks for the tip about the subTest.
Following your advice, I have separate each part of the use-case in separated test methods.
It is indeed easier to read.
About "don't rely on the default values set in the setUpClass", I think it is not entirely feasible in this case as we test here that when the logo has never been set, we get the default logo (and the super user company is already created, even outside of the setUpClass). 
So except for the default logo, the logo initialization is done at the beginning of each test method.